Assets/MixedRealityToolkit.Services/DiagnosticsSystem/DiagnosticsSystemVoiceControls.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Utilities;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Diagnostics
+{
+    /// <summary>
+    /// Class that listens for and acts upon diagnostic system voice commands.
+    /// </summary>
+    public class DiagnosticsSystemVoiceControls : MonoBehaviour, IMixedRealitySpeechHandler
+    {
+        private IMixedRealityDiagnosticsSystem diagnosticsSystem = null;
+
+        private IMixedRealityDiagnosticsSystem DiagnosticsSystem
+        {
+            get
+            {
+                if (diagnosticsSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityDiagnosticsSystem>(out diagnosticsSystem);
+                }
+                return diagnosticsSystem;
+            }
+        }
+
+        private IMixedRealityInputSystem inputSystem = null;
+
+        private IMixedRealityInputSystem InputSystem
+        {
+            get
+            {
+                if (inputSystem == null)
+                {
+                    MixedRealityServiceRegistry.TryGetService<IMixedRealityInputSystem>(out inputSystem);
+                }
+                return inputSystem;
+            }
+        }
+
+        bool registeredForInput = false;
+
+        private void OnEnable()
+        {
+            if (!registeredForInput)
+            {
+                if (InputSystem != null)
+                {
+                    InputSystem.Register(gameObject);
+                    registeredForInput = true;
+                }
+            }
+        }
+
+        private void OnDestroy()
+        {
+            if (registeredForInput)
+            {
+                InputSystem.Unregister(gameObject);
```

Comment:
This might be desired behavior. But, there seems to be some asymmetry between the InputSystem.Register andInputSystem.Unregister calls. Should this occur in OnDisable()? Or, should it be Start() and OnDestory()?

Reply:
interesting question. on disable may be the right place (as well as destroy). start may be too early, but it wouldnt hurt to add it there too

Reply:
would you like the asymmetry fixed before merging?

Reply:
Functionally it seems perfect to me! So feel free to merge. I just have found myself making those symmetry mistakes so thought I would bring it up. I tend to follow the OnEnable/OnDisable pattern for all registrations so that users can just enable/disable objects to subscribe/unsubscribe.

Reply:
Spoke with Cameron. if the asymmetry proves to be an issue, it will be bugged and updated.